src/Pages/HomePage/HomePage.js

```diff
@@ -1,21 +1,79 @@
-import React  from 'react';
+import React , { useState }  from 'react';
 import { SideBar, CategoryBar } from '../../components/index';
-import { useVideos } from '../../context/index';
+import { useVideos , useHistoryContext , useWatchlaterContext , usePlayListContext, useToastContext, useUserAuth } from '../../context/index';
 import './HomePage.css';
 import { Link } from 'react-router-dom';
-import { addVideoInHistory } from '../../api-calls/api-calls';
-import { useHistoryContext } from '../../context/HistoryContext';
+import { addVideoInHistory , addToWatchLater, removeFromWatchLater , createPlayList , addVideoToPlaylist , removeVideoFromPlaylist } from '../../api-calls/api-calls';
 
 export function HomePage() {
   const { historyVideosDispatch } = useHistoryContext();
+  const { watchLaterDispatch } = useWatchlaterContext();
+  const { filteredVideos , setFilteredVideos , setVideos } = useVideos();
+  const [showModal , setShowModal ] = useState(false);
+  const [playlistName , setPlaylistName ] = useState('');
+  const { playListState , playListDispatch } = usePlayListContext();
+  const { isLogIn } = useUserAuth();
+  const notify = useToastContext();
+  let playlists;
+    if(playListState.payload === undefined || playListState.payload === 'none'){
+      playlists = [];
+    }else{
+      playlists = playListState.payload.playlist.data.playlists;
+    }
+
+  const updatePlayList = async(e, playlistId , videoData) => {
```

Comment:
Suggestion: you can extract all actions and put inside custom hook to avoid repetition of code